lib/structures/Command.js

```diff
@@ -1,6 +1,18 @@
 "use strict";
+const Precondition = require("./Precondition");
+const Middleware = require("./Middleware");
 
 module.exports = class Command {
+  /**
+   *
+   * @param {Object} options
+   * @param {String[] | String} options.names
+   * @param {String[] | String} options.permissions
+   * @param {Number} options.cooldown
+   * @param {Precondition[] | Precondition} options.preconditions
+   * @param {Middleware[] | Middleware} options.middleware
```

Comment:
`options.preconditions` and `options.middleware` should be Strings

Reply:
This was because I was under the impression that `options.preconditions` & `options.middleware` would be `Precondition[]` & `Middleware[]` respectively.
Fixed.